### PR TITLE
Fix duplicate subscription + changed some namespaces + fixed issue wi…

### PR DESCRIPTION
**Pull request by Krayt78**
…th wearableNFT

### CODE REVIEW

File: Assets/MirageSDK/Examples/Scripts/WearableNFTExample/WearableNFTExample.cs [hunk 1]
```diff
@@ -98,23 +98,38 @@ private async UniTask GameItemSetApproval()
 			UpdateUILogs($"Game Items approved. Hash : {transactionHash}");
 		}
 
-		private async UniTask<BigInteger> GetBalance()
+		private async UniTask<BigInteger> GetBalanceERC721(IContract contract)
```
[the remark this answers is not in the document]
Comment (Krayt78) [reply Assets/MirageSDK/Examples/Scripts/WearableNFTExample/WearableNFTExample.cs:101]: Should i use our new ContractFunctions for that ? the script is located in miragesdk/script/core/utils

File: Assets/MirageSDK/Scripts/Core/Utils/ContractFunctions.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using Cysharp.Threading.Tasks;
 using MirageSDK.Core.Infrastructure;
 using MirageSDK.Examples.Scripts.ContractMessages;
```
[the remark this answers is not in the document]
Comment (Vukiz) [reply Assets/MirageSDK/Scripts/Core/Utils/ContractFunctions.cs:4]: Scripts folder should stay, but scripts folder should not produce namespace

File: Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs [hunk 3]
```diff
@@ -99,28 +82,38 @@ protected override async void Awake()
 
 			base.Awake();
 
-			if (connectOnAwake)
-			{
```
Comment (Vukiz) [inline Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs:103]: do not remove braces, braces should be enforced by default. Forcing that causes much less errors and easier for maintenance

File: Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs [hunk 4]
```diff
@@ -99,28 +82,38 @@ protected override async void Awake()
...
+		}
+
+		private async void OnDestroy()
+		{
+			await SaveOrDisconnect();
 		}
 
+		private async void OnApplicationPause(bool pauseStatus)
+		{
+			if (pauseStatus)
```
Comment (Vukiz) [inline Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs:100]: please force braces everywhere on this file and check you IDE settings as this should be your default behaviour.

File: Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs [hunk 5]
```diff
@@ -130,18 +123,15 @@ public async Task<WCSessionData> Connect(CancellationToken cancellationToken = d
...
-						{
-							await Session.Transport.Close();
-						}
+						else if (Session.TransportConnected) await Session.Transport.Close();
 					}
 					else if (!Session.Connected && !Session.Connecting)
 					{
 						SetupDefaultWallet().Forget();
 
+						/// -= Removes the event if it has been already added, this prevents multiple firing of the event
```
[the remark this answers is not in the document]
Comment (Vukiz) [reply Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs:133]: Also we are duplicating ALL the events, and you only covered one case. I think you should reconsider this and place it somewhere more appropriate, and keep all subscriptions clear

**Review by Vukiz — CHANGES_REQUESTED** [review]

File: Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs (hunk 4 above)
Comment (Krayt78) [reply Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs:100]: yeah its my ide that does it i will check how to fix that

File: Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Krayt78) [reply Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs:133]: i wasnt sure if i needed to change all of them or not, I will just make a generic unsubscribe function and call it instead

**Review by Vukiz — APPROVED** [review]

**Pull request merged**